### PR TITLE
PROD-4098 -> PROD-4246 - Improve moderation for media

### PR DESCRIPTION
**Pull request by krunal-buddyboss**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-4098

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

[PROD-4098]: https://buddyboss.atlassian.net/browse/PROD-4098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

File: src/bp-templates/bp-nouveau/includes/video/ajax.php [hunk 1]
```diff
@@ -1260,6 +1260,9 @@ function bp_nouveau_ajax_video_get_video_description() {
 				'include'  => $video_id,
 				'album_id' => 'existing-video',
 			);
+			if( ! empty( $video->group_id ) ) {
```
[the remark this answers is not in the document]
Comment (krunal-buddyboss) [reply src/bp-templates/bp-nouveau/includes/video/ajax.php]: Fixed it

File: src/bp-moderation/classes/suspend/class-bp-suspend-video.php [hunk 2]
```diff
@@ -319,6 +319,10 @@ public function update_where_sql( $where_conditions, $args = array() ) {
 			return $where_conditions;
 		}
 
+		if( ( isset( $args['scope'] ) && 'groups' === $args['scope'] ) || ! empty( $args['group_id'] ) ) {
```
[the remark this answers is not in the document]
Comment (krunal-buddyboss) [reply src/bp-moderation/classes/suspend/class-bp-suspend-video.php]: Fixed it

File: src/bp-moderation/classes/suspend/class-bp-suspend-document.php [hunk 3]
```diff
@@ -231,6 +231,10 @@ public function update_where_sql( $where_conditions, $args = array() ) {
 			return $where_conditions;
 		}
 
+		if( ( isset( $args['scope'] ) && 'groups' === $args['scope'] ) || ! empty( $args['group_id'] ) ) {
```
[the remark this answers is not in the document]
Comment (krunal-buddyboss) [reply src/bp-moderation/classes/suspend/class-bp-suspend-document.php]: FIxed it

File: src/bp-moderation/classes/suspend/class-bp-suspend-media.php [hunk 4]
```diff
@@ -236,6 +236,10 @@ public function update_where_sql( $where_conditions, $args = array() ) {
 			return $where_conditions;
 		}
 
+		if( ( isset( $args['scope'] ) && 'groups' === $args['scope'] ) || ! empty( $args['group_id'] ) ) {
```
[the remark this answers is not in the document]
Comment (krunal-buddyboss) [reply src/bp-moderation/classes/suspend/class-bp-suspend-media.php]: Fixed it

File: src/bp-moderation/classes/class-bp-moderation-members.php [hunk 5]
```diff
@@ -191,8 +191,8 @@ public function restrict_member_profile() {
 	public function bp_core_get_user_domain( $domain, $user_id ) {
 		$username_visible = isset( $_GET['username_visible'] ) ? sanitize_text_field( wp_unslash( $_GET['username_visible'] ) ) : false;
 
-		if ( empty( $username_visible ) && bp_moderation_is_user_blocked( $user_id ) ) {
-			return '';
+		if ( empty( $username_visible ) && ( bp_moderation_is_user_blocked( $user_id ) || bb_moderation_is_user_blocked_by( $user_id ) ) ) {
+			return ' ';
```
[the remark this answers is not in the document]
Comment (krunal-buddyboss) [reply src/bp-moderation/classes/class-bp-moderation-members.php]: As this space needs to allow the function bp_core_get_userlink to work, I added a comment for the same as well.

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**